lumicks/pylake/detail/confocal.py

```diff
@@ -6,22 +6,48 @@
 from .mixin import PhotonCounts
 from .mixin import ExcitationLaserPower
 from .image import reconstruct_image_sum, reconstruct_image, save_tiff, ImageMetadata
+from .utilities import could_sum_overflow
+
+
+def _int_mean(a, total_size, axis):
+    """Compute the `mean` for `a`.
+
+    When the mean can safely be computed for the entire block, we simply do so in a single step.
+    The early return will actually be called in most cases. When an overflow would occur, we
+    split the blocks to sum over recursively until the individual blocks can be summed without
+    incurring an overflow."""
+    if not could_sum_overflow(a, axis):
+        return np.sum(a, axis) // total_size
+
+    b = a if axis is None else np.swapaxes(a, axis, 0)
+    n = b.shape[0] // 2
```

Comment:
I've added:
```
    # Swap axis is used so the dimension we average over is the first. This makes it easier to
    # divide it into blocks and sum over them.
```
It's faster than `np.take` since it doesn't generate a copy. And it is a lot cleaner than slicing with a list of `Slice`s. It also makes things generalize nicely to 1D.

Reply:
Yup, those are exactly the reasons. Conventional slicing syntax is really messy for this and most other `numpy` functions return copies. The fact that `swapaxes()` just returns a different view to the same data is really neat!